cluster/addons/metrics-server/metrics-server-deployment.yaml

```diff
@@ -65,6 +65,13 @@ spec:
         - containerPort: 443
           name: https
           protocol: TCP
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            add:
+            - NET_BIND_SERVICE
+            drop:
+            - all
```

Comment:
did adding this change anything? I thought that if `allowPrivilegeEscalation` and `capabilities` were absent from the pod spec, the container had NET_BIND_SERVICE by default

cc @vinayakankugoyal

Reply:
cam this be a problem dependon on the type of node the pods has been scheduled?

Reply:
I... don't know?

I don't really object to this, since I think it's a no-op, I'm just trying to avoid churn / cargo-culting things that don't actually help

Reply:
I don't know either, actually I'm trying to find out how to know if this really solves the issue, but I can't find anything on the presubmits jobs

Reply:
can this be a problem that the job uses docker and docker doesn't respect capabilites?
```
      - --env=KUBE_CONTAINER_RUNTIME=docker
```

Reply:
> can this be a problem that the job uses docker

pull-kubernetes-e2e-gce-ubuntu-containerd doesn't use containerd?

> and docker doesn't respect capabilites?

docker does respect capabilities, and includes NET_BIND_SERVICE by default if unspecified

Reply:
as a nit, use `ALL` if we're going to add this

Reply:
> pull-kubernetes-e2e-gce-ubuntu-containerd doesn't use containerd?
> 
> >

I think that the difference is in the kubelet codepath:
* _containerd_ use CRI containerd,
* _docker_ uses dockershim

Reply:
isn't this running as root? Why is this needed?

Reply:
@vinayakankugoyal permission denied when trying to bind https://github.com/kubernetes/kubernetes/issues/103708#issuecomment-880599741

Reply:
aah do you have file caps on the binary? my guess is that you may have to remove allowPrivilegeEscaltion:false

Reply:
yes, there are file capabilities on the binary

Reply:
removed `allowPrivilegeEscaltion:false` and set ALL

Reply:
@aojea why did you add it back in your most recent changes?

Reply:
wrong rebase Xd